### DISCUSSION TITLE
Element's horizontal scroll position snaps back when scrolling the document vertically

### DISCUSSION THREAD

**Discussion by jenky** [hirasso/scrollmirror#29] [Q&A] [answered]
### What did you expect? 🧐

It should not snap back to the start when scrolling in the opposite direction.

### What actually happened? 😵‍💫

I have 2 elements with overflow-x and using scroll mirror to sync the scroll horizontally. It works, but as soon as I scroll vertically, the horizontal scroll snaps back to the start

```js
{ vertical: false, horizontal: true }
```

https://github.com/user-attachments/assets/b5ab3e82-238f-4cd9-b1dd-ebe7c1de6808

### What browsers are you seeing the problem on? 🧭

Chrome

### URL to minimal reproduction 🔗

N/A

### Checked all these? 📚

- [X] I have read the [usage instructions](https://github.com/hirasso/scrollmirror#usage-example)
- [X] I have checked [discussions](https://github.com/hirasso/scrollmirror/discussions) and [existing issues](https://github.com/issues?q=is%3Aissue+repo%3Ahirasso%2Fscrollmirror+) for related problems
- [X] I agree to follow this project's [Code of Conduct](https://github.com/hirasso/scrollmirror/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by hirasso** [comment]
Hi @jenky , thanks for your report! Could you create a minimal reproduction of the bug? You can fork this pen on codepen: https://codepen.io/rassohilber/pen/JjxwJpo

**Comment by jenky** [comment]
@hirasso Here is the reproduction https://codesandbox.io/p/sandbox/ptscft

**Comment by hirasso** [comment, marked as answer]
Thank you @jenky!

In your code, you are also adding `:root` to the mirrored elements. That's the reason why when scrolling the document vertically, the horizontal scroll container correctly snaps back to zero, as the document's `scrollLeft` is zero. Removing `:root` fixes the problem:

```diff
- document.querySelectorAll(":root, .comparison-container", {
+ document.querySelectorAll(".comparison-container", {
  vertical: false,
  horizontal: true,
})
```

If you like this library, please give it a star! It dearly needs some 🙂

**Reply by hirasso** [reply]
@jenky did this answer your question?

**Reply by jenky** [reply]
Ah yep, thank you so much. My notification directed me to the original issue that is closed, and I thought I couldn't leave a message. I didn't realize it had been converted to a discussion.
